src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -441,8 +441,72 @@ void bulk_load_service::try_resend_bulk_load_request(const std::string &app_name
 void bulk_load_service::handle_app_downloading(const bulk_load_response &response,
                                                const rpc_address &primary_addr)
 {
-    // TODO(heyuchen): TBD
-    // called by `on_partition_bulk_load_reply` when app status is downloading
+    const std::string &app_name = response.app_name;
+    const gpid &pid = response.pid;
+
+    if (!response.__isset.total_download_progress) {
```

Comment:
Meta send partition_bulk_load_request to primary, primary report download progress, meta receive response and resend request to primary until bulk load finished or failed. As a result, if primary don't report progress in this response, it's okay, it will be reported in further response.
It is possible for replica not report progress, it's not a normal case, so I print warning log here.

Reply:
Please notice the reason why "total_download_progress" is unset in your warning log.

> As a result, if primary don't report progress in this response,
I still don't see why this could happen. Does that mean the replica is blocked? Or the bulk-load process is not yet started?

"As a result, if primary don't report progress in this response, it's okay, it will be reported in further response."
If so, we can clearly see from the download processes, that some partition perhaps blocks and the process is long-time not updated. No warning log needed.

Reply:
It will happen during error handling, consider the case:
when error happened during ingestion or finish, app status has turned to downloading, it may handle the response reported from replica partition status is ingestion or finish, whose progress is not set.